templates/main.tftpl

```diff
@@ -1 +1,29 @@
-#This file is where the code for your application will be in the future.
\ No newline at end of file
+resource "azurerm_resource_group" "network" {
+  name     = "rg-$${local.application_name}Network-$${local.stage}-$${local.env_num}"
+  location = local.default_location
+  tags     = local.tags
+}
+
+/* resource "azurerm_resource_group" "security" {
+  name     = "rg-$${local.application_name}Security-$${local.stage}-$${local.env_num}"
+  location = local.default_location
+  tags     = local.tags
+} */
+
+/* resource "azurerm_resource_group" "app" {
+  name     = "rg-$${local.application_name}Application-$${local.stage}-$${local.env_num}"
+  location = local.default_location
+  tags     = local.tags
+} */
+
+/* resource "azurerm_resource_group" "database" {
+  name     = "rg-$${local.application_name}Database-$${local.stage}-$${local.env_num}"
+  location = local.default_location
+  tags     = local.tags
+} */
+
+/* resource "azurerm_resource_group" "fileshare" {
+  name     = "rg-$${local.application_name}Files-$${local.stage}-$${local.env_num}"
+  location = local.default_location
+  tags     = local.tags
+} */
```

Comment:
Commented code is generally considered "code smell" since in the future we might forget why it was needed or why it was removed. Here we could use template variables to conditionally deploy these RGs if necessary or just remove them if they are commented out anyway.

Reply:
Solved deleting comment and managed by variable maps of resources groups